vcd/resource_vcd_catalog_media.go

```diff
@@ -283,12 +319,20 @@ func createOrUpdateMediaItemMetadata(d *schema.ResourceData, meta interface{}) e
 
 	vcdClient := meta.(*VCDClient)
 
-	org, err := vcdClient.GetAdminOrgFromResource(d)
-	if err != nil {
-		return fmt.Errorf(errorRetrievingOrg, err)
+	var catalog *govcd.Catalog
+	var err error
+	var orgName string
+	catalogName := d.Get("catalog").(string)
+	catalogId := d.Get("catalog_id").(string)
+	if catalogId == "" {
+		orgName, err = vcdClient.GetOrgNameFromResource(d)
+		if err != nil {
+			return fmt.Errorf("error retrieving Org name for vcd_catalog_media: %s", err)
+		}
+		catalog, err = vcdClient.Client.GetCatalogByName(orgName, catalogName)
+	} else {
+		catalog, err = vcdClient.Client.GetCatalogById(catalogId)
```

Comment:
As currently "catalog_id" and "catalog" are both Optional, both can be "". Maybe we can aid users to avoid this situation by adding AtLeastOneOf?